openshift-setup/router-template.cfg

```diff
@@ -0,0 +1,654 @@
+    {{- if (eq $cfg.TLSTermination "") }}
+
+# Plain http backend
+backend be_http:{{$cfgIdx}}
+    {{- else if (eq $cfg.TLSTermination "edge") }}
+
+# Plain http backend but request is TLS, terminated at edge
+backend be_edge_http:{{$cfgIdx}}
+    {{ else if (eq $cfg.TLSTermination "reencrypt") }}
+
+# Secure backend which requires re-encryption
+backend be_secure:{{$cfgIdx}}
+    {{- end }}{{/* end chceck for router type */}}
+  mode http
+  option redispatch
+  option forwardfor
+
+    {{- with $balanceAlgo := firstMatch "roundrobin|leastconn|source" (index $cfg.Annotations "haproxy.router.openshift.io/balance") (env "ROUTER_LOAD_BALANCE_ALGORITHM") }}
+  balance {{ $balanceAlgo }}
+    {{- else }}
+  balance {{ if gt $cfg.ActiveServiceUnits 1 }}roundrobin{{ else }}leastconn{{ end }}
+    {{- end }}
+    {{- with $ip_whiteList := firstMatch $cidrListPattern (index $cfg.Annotations "haproxy.router.openshift.io/ip_whitelist") }}
+  acl whitelist src {{ $ip_whiteList }}
+  tcp-request content reject if !whitelist
+    {{- end }}
+    {{- with $value := firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout")}}
+  timeout server  {{$value}}
+    {{- end }}
+
+    {{- if isTrue (index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections") }}
+  stick-table type ip size 100k expire 30s store conn_cur,conn_rate(3s),http_req_rate(10s)
+  tcp-request content track-sc2 src
+      {{- if (isInteger (index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections.concurrent-tcp")) }}
+  tcp-request content reject if { src_conn_cur ge  {{ index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections.concurrent-tcp" }} }
+      {{- else }}
+  # concurrent TCP connections not restricted
+      {{- end }}
+
+      {{- if (isInteger (index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections.rate-tcp")) }}
+  tcp-request content reject if { src_conn_rate ge {{ index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections.rate-tcp" }} }
+      {{- else }}
+  #TCP connection rate not restricted
+      {{- end }}
+
+      {{- if (isInteger (index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections.rate-http")) }}
+  tcp-request content reject if { src_http_req_rate ge {{ index $cfg.Annotations "haproxy.router.openshift.io/rate-limit-connections.rate-http" }} }
+      {{- else }}
+  #HTTP request rate not restricted
+      {{- end }}
+    {{- end }}
+
+  timeout check 5000ms
+  http-request set-header X-Forwarded-Host %[req.hdr(host)]
+  http-request set-header X-Forwarded-Port %[dst_port]
+  http-request set-header X-Forwarded-Proto http if !{ ssl_fc }
+  http-request set-header X-Forwarded-Proto https if { ssl_fc }
+
+  {{/* set X-LAGOON header if it is not already */}}
+  acl h_xlagoon_exists res.hdr(X-LAGOON) -m found
+  http-response set-header X-LAGOON {{env "HOSTNAME" }}>{{$cfgIdx}} unless h_xlagoon_exists
```

Comment:
any chance that we can add the HOSTNAME all the time even when the header already exists?